app/controllers/pets_controller.rb

```diff
@@ -12,11 +12,18 @@ def new
 
   def create
     shelter = Shelter.find(params[:id])
-    pet = shelter.pets.create!(pet_params)
+    pet = shelter.pets.create(pet_params)
 
-    redirect_to "/shelters/#{shelter.id}/pets"
+    if pet.save
+      flash[:message] = "#{pet.name} has been created!"
+      redirect_to "/shelters/#{shelter.id}/pets"
+    else
+      flash[:message] = "You must fill out #{empty_params}"
```

Comment:
this message now informs users in exactly which params have not been entered

Reply:
nice!